File: 108.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return dfs(nums, 0, nums.length-1);
    }

    public TreeNode dfs(int[] nums, int l, int r) {
        if(l > r) return null;
        int m = (l+r)/2;
        TreeNode cur = new TreeNode(nums[m]);
        cur.left = dfs(nums, l, m-1);
        cur.right = dfs(nums, m+1, r);

        return cur;
    }
}

File: GO/108.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sortedArrayToBST(nums []int) *TreeNode {
	return dfs(nums, 0, len(nums)-1)
}

func dfs(nums []int, l, r int) *TreeNode {
	if l > r {
		return nil
	}
	m := (l + r) / 2
	cur := &TreeNode{Val: nums[m]}
	cur.Left = dfs(nums, l, m-1)
	cur.Right = dfs(nums, m+1, r)

	return cur
}